### PR TITLE
[PUB-1562] Replace upgrade to pro paths by pro trial

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description

Add Trial Pro path for free users instead of Upgrade to Pro.
Places:
- Upgrade to Pro CTA on Tab navbar
- Menu item in Appshell
- CTA in Locked Profiles Message

Extra tasks:
- Add notification messages to notifications.json, to display success and error messages coming from buffer-web when starting a Trial.
- Replace links to pro and business pages by a link to the pricing page (Spencer did it across the products, but missed one spot in the composer).

## Context & Notes

Users in a free plan, that never had a Pro Trial and are not in any other trial, are now encouraged to take a Pro trial, hoping that it will increase the conversion rate.
[PUB-1562](https://buffer.atlassian.net/browse/PUB-1562)

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/composer/interfaces/buffer-publish.jsx [hunk 1]
```diff
@@ -175,11 +176,17 @@ events.on('saved-drafts', () => {
...
 
-events.on('start-trial', (message) => {
+events.on('start-trial', ({message, removeScope}) => {
   // reformat new userData
   const userData = DataImportUtils.formatUserData(null, { userData: message });
   AppInitActionCreators.resetUserData(userData);
-  // Trigger the comment toggle
+
+  if (removeScope) {
+    NotificationActionCreators.removeAllNotificationsByScope(removeScope);
```
Comment (albennett) [inline packages/composer/interfaces/buffer-publish.jsx]: Curious what notifications would be showing?

File: packages/composer/composer/action-creators/AppActionCreators.js [hunk 2]
```diff
@@ -206,11 +207,19 @@ const AppActionCreators = {
...
-                action: () => {
+                action: (event) => {
                   const { environment } = AppStore.getMetaData();
                   if (isFreeUser) {
                     if (canStartProTrial) {
-                      window.open(`${bufferOrigins.get(environment)}/billing/start-trial?trialType=pro`);
+                      if (event && event.target && !event.target.disabled) {
+                        event.target.disabled = true;
+                      }
+                      AppActionCreators.triggerInteraction({
```
Comment (albennett) [inline packages/composer/composer/action-creators/AppActionCreators.js]: You'd probably want to do a similar case where a user sees the upgrade modal if they aren't eligible for a trial, right? If that's the case, you may want to add an eng task for it. I'm currently looking into how we can upgrade without a refresh.

File: packages/trial/middleware.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import {
```
Comment (albennett) [inline packages/trial/middleware.js]: Thanks for making this logic more shareable with the trials package. 💯

File: packages/composer/interfaces/buffer-publish.jsx (hunk 1 above)
Comment (karelm) [reply packages/composer/interfaces/buffer-publish.jsx]: Great question @albennett, this is notification in the composer itself, so its not presently working via that route... but if I can move notifications to all exist in the context of the publish notifications package, hopefully we can start removing logic such as the notification store in composer land :) 

Here is what it looks like in action:
http://hi.buffer.com/baefd38895bf

**Pull request merged**